models/User.go

```diff
@@ -857,14 +864,30 @@ func PlaceAskOrder(userId uint32, ask *Ask) (uint32, error) {
 
 	l.Info("Saved OrderFeeTransaction for bid %d", ask.Id)
 
+	// Going to reserve stocks for this order
+	placeOrderTransaction := GetTransactionRef(
```

Comment:
If you're using this function only like this, why make the caller pass `PlaceOrderTransaction`, 0, 0 etc? Just userID, ask.StockID, -ask.StockQuantity should be enough no?